### DISCUSSION TITLE
Why is Laravel (migration) float create a double type column?

### DISCUSSION THREAD

**Discussion by ccec1t91077** [laravel/framework#47080] [Q&A]
Hi everyone, I'd like to know why Laravel `float()` create a double type column?

I've view a lot of discussions, but I don't see any framework member giving a reason, and there seems no conclusion, I know there have some solutions can work around for this, but I wonder the reason why.

If there's a new project, I definitely choose the double type, but sometimes I would need to upgrade some none framework projects to Laravel, there will be a lot of data exists, I'll like to make same type column to make sure the data won't lost.

This behover definitely seems incorrect on me (because there does have an `double()` to create a double type column), even if there have precision issue, it should still create a float instead of double, it could have been an official solution, if someone need it.

Well... can someone tell me the reason or give a reasonable reason for me, I'm curious.

I've view the issue below and more.
[#7298](https://github.com/laravel/framework/issues/7298)
[#9103](https://github.com/laravel/framework/issues/9103)
[#3151](https://github.com/laravel/framework/issues/3151#issuecomment-464381733)

**Comment by hafezdivandari** [comment]
Fixed on Laravel 11.x: #48861
